### PR TITLE
Closes #90 and #126

### LINKED ISSUE #90: Base line match - similar names, recall can be improved
Similar names match produces less matches than the perfect one.

ISSUES:
• Tokens foreach id are stored into a set. Set does not guarantee order of the items, so the conversion to string is not deterministic.
• Tokens are joined to build a string, then it's compared to the other database strings. This enables to do a strict equality comparison. Though, a lot of links are not done due to the presence of Arabic characters in Wikidata and not available on the other database.

POSSIBILE SOLUTION:
• treating tokens comparison in their set form. The rule could be: `if a set of tokens is contained in another set of tokens, they are a match`

### LINKED ISSUE #126: Investigate external libraries for full-text indices on MariaDB
SQL Alchemy does not natively support creation of fullt-text indices.
Investigate https://github.com/mengzhuo/sqlalchemy-fulltext-search

### PR DESCRIPTION
**Pull request by MaxFrax**
Closes #90 
Closes #126

### CODE REVIEW

File: soweego/linker/linking_strategies.py [hunk 1]
```diff
@@ -47,90 +50,134 @@ def baseline(source, target, strategy, output_dir):
...
-        for string, identifier in dataset.items():
-            merged[string].append(identifier)
-    for string, identifiers in merged.items():
-        if len(identifiers) > 1:
-            LOGGER.debug("'%s': it's a match! %s -> %s",
-                         string, identifiers[0], identifiers[1])
-            matched[identifiers[0]] = identifiers[1]
+
+    for label, qid in source_dataset.items():
+        for res in session.query(target_entity).filter(target_entity.name == label).all():
```
Comment (marfox) [inline soweego/linker/linking_strategies.py]: - TODO: move `connect_to_db` to `utils`;
- should use a safer construct here, as per `data_gathering.py`, i.e.:
```py
def connect_to_db():
    db_manager = DBManager()
    session = db_manager.new_session()
    return session

def DO_STUFF():
    session = connect_to_db()
    result = None
    try:
        result = RUN SOME QUERY HERE
        session.commit()
    except:
        session.rollback()
        raise
    finally:
        session.close()

    if not result:
        return None
    return PARSE_RESULT_FUNCTION(result)

```

File: soweego/linker/linking_strategies.py [hunk 2]
```diff
@@ -47,90 +50,134 @@ def baseline(source, target, strategy, output_dir):
...
+
+        to_exclude.clear()
+
+        tokenized = tokenize(label)
+        if len(tokenized) <= 1:
+            continue
+
+        boolean_search = ' '.join(map('+{0}'.format, tokenized))
+        natural_search = ' '.join(tokenized)
+        session = db_manager.new_session()
```
Comment (marfox) [inline soweego/linker/linking_strategies.py]: Same comment as above

File: soweego/linker/linking_strategies.py [hunk 3]
```diff
@@ -47,90 +50,134 @@ def baseline(source, target, strategy, output_dir):
...
+
+        # NOTICE: sets of size 1 are always exluded
+        # Looks for sets equal or bigger containing our tokens
+        ft_search = target.tokens.match(boolean_search)
+        for res in session.query(target).filter(ft_search).all():
```
Comment (marfox) [inline soweego/linker/linking_strategies.py]: **N.B.:** command for full-text index query.

Comment (MaxFrax) [reply soweego/linker/linking_strategies.py]: @marfox Can you please tell me more about this?

**Pull request merged**